rpc/gnmi/gnmi-union_replace.md

```diff
@@ -0,0 +1,327 @@
+Although the concept is simple enough, there are portions of the configuration which may not be modeled in OpenConfig.
+
+## 2. Definitions
+
+### 2.1 Network Operating System (NOS)
+
+The operating system running on a network device which provides the management interfaces.
+
+### 2.2 OpenConfig Configuration (OC)
+
+OC is defined as a set of OpenConfig working group approved YANG models. The encoding for OC is defined in the gNMI specification.  The origin for a path in gNMI is by default (empty) OC or may be explicitly stated as origin:openconfig.  Vendor augmentations to the OC models should be regarded as being part of origin:openconfig.
+
+### 2.3 Native YANG Configuration (NY)
+
+NY is defined as a set of YANG models which are proprietary to a network operating system (NOS). The origin of this configuration must be `origin:<nos_native>`.  The NOS determines the value for `<nos_native>`.  Example values for `<nos_native>` might include:  `xros_native`, `junos_native`, `srlinux_native`, `sonic_native` or any string which indicates the NOS name and the native designation.
+
+### 2.4 Native CLI Configuration (CLI)
+
+CLI configuration is defined as the NOS command line interface text used for configuration.  In gNMI, this is encoded as ASCII text.  The format of the ASCII text is agreed to out of band from  gNMI.  The origin of this configuration must be `origin:<nos_cli>`.  Example values might include: `xros_cli`, `junos_cli`, `srlinux_cli` or any string which indicates the NOS name and the CLI designation.
+
+### 2.5 Startup configuration
+
+The configuration a device will load and use as the running configuration after the base operating system has booted.
+
+### 2.6 Running or Active configuration
+
+The configuration which is in effect on the device.
+
+### 2.7 Boot Config
+
+The operating system (Linux) boot configuration which provides key-value data when booting the kernel.  This is not the device startup configuration.  Boot config items are managed by bootz.
+
+### 2.8 bootz
+
+[bootz](https://github.com/openconfig/bootz) describes structured data and API for taking a device from factory defaults to a ready for production state.  bootz is intended to replace secure zero touch provisioning (sZTP) solutions.
+
+Bootz defines partitions or namespaces for NOS configuration items and an order for applying these namespaces to the NOS.  This is described briefly later in this document and in detail at the [bootz](https://github.com/openconfig/bootz) repository.
+
+### 2.9 gNSI
+
+gNSI requires modeled config paths to be disabled or restricted from OC and NY paths.  If these gNSI controlled paths are attempted to be set using a gNMI `SetRequest`, the `SetRequest` must be rejected with an error.
```

Comment:
It's not clear to me what this first sentence means. gNSI allows a policy to be set that can control access to these paths, but doesn't require disabled or restricted by default AFAICS. Can you clarify?

Reply:
Updated with reference to pathz and linked to bootz's definition of interoperation with gNSI.